packages/transport/src/core/types/PasswordProtection.ts

```diff
@@ -31,4 +32,15 @@ export class PasswordProtection extends Serializable implements IPasswordProtect
             salt: this.salt
         });
     }
+
+    private matchesCreationParameters(compare: PasswordProtectionCreationParameters): boolean {
+        return this.passwordType === compare.passwordType && this.password === compare.password;
+    }
+
+    public matchesInputForNewPassword(compare: { password: string; passwordIsPin?: true } | undefined): boolean {
+        const compareCreationParameters = PasswordProtectionCreationParameters.create(compare);
+        if (!compareCreationParameters) return false;
+
+        return this.matchesCreationParameters(compareCreationParameters);
```

Comment:
as this function calls the other private function the private function has to be BELOW this. You can read more about this in our coding guidelines.